### PR TITLE
[Rust] Added doc comments for Function, Memory, and Table

### PR DESCRIPTION
**Pull request by apepkuss**
This PR is to add the necessary descriptions for the Function, Memory, and Table structs.

### CODE REVIEW

File: bindings/rust/wasmedge-sys/src/instance/memory.rs [hunk 1]
```diff
@@ -127,13 +284,34 @@ impl Drop for Memory {
...
+    /// specifies the min size (also, initial size) of the memory, while the end value specifies
+    /// the max size allowed to grow.
+    ///
+    /// # Errors
+    ///
+    /// If fail to create a [`MemType`], then an error is returned.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/memory.rs:310]: Why does this example ignore?

File: bindings/rust/wasmedge-sys/src/instance/memory.rs [hunk 2]
```diff
@@ -37,23 +70,83 @@ impl Memory {
...
+    /// - `offset` specifies the data start offset in the [`Memory`].
+    ///
+    /// - `len` specifies the requested data length.
+    ///
+    /// # Errors
+    ///
+    /// If the `offset` + `len` is larger than the data size in the [`Memory`], then an error is returned.
+    ///
+    /// # Examples
+    ///
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/memory.rs:86]: There is no example here?

File: bindings/rust/wasmedge-sys/src/instance/memory.rs [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+    /// - `limit` specifies a inclusive range of the page count in the [`Memory`]. For example,
+    /// (10..=20) means the lower bound (inclusive) of the page count is 10 ,while the upper bound (inclusive) 20.
+    ///
+    /// # Errors
+    ///
+    /// If fail to create a [`Memory`], then an error is returned.
+    ///
+    /// # Example
+    ///
+    /// ```ignore
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/memory.rs:33]: Why does this example ignore?

File: bindings/rust/wasmedge-sys/src/instance/global.rs [hunk 4]
```diff
@@ -125,7 +125,7 @@ impl Drop for Global {
 
 /// Struct of WasmEdge GlobalType.
 ///
-/// `GlobalType` specifies whether a global variable is immutable or mutable, and also the data type of the value stored in a `Global`.
+/// A [`GlobalType`] classifies a global variable, which hold a value and can either be mutable or immutable.
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/global.rs:128]: `which holds a value`.

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/instance/global.rs (hunk 4 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/global.rs:128]: Accepted. Will refine the statement, thanks a lot!

File: bindings/rust/wasmedge-sys/src/instance/memory.rs (hunk 3 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/memory.rs:33]: The "ignore" here notifies cargo not to do doc test against this piece of code. Thanks!

File: bindings/rust/wasmedge-sys/src/instance/memory.rs (hunk 2 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/memory.rs:86]: Accepted. Removed the Example tag. Thanks!

File: bindings/rust/wasmedge-sys/src/instance/memory.rs (hunk 1 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/memory.rs:310]: The "ignore" here notifies cargo not to do doc test against this piece of code. Thanks!

**Comment by apepkuss** [comment]
@hydai the review issues have been fixed, please review the latest changes again. Thanks a lot!

File: bindings/rust/wasmedge-sys/src/instance/function.rs [hunk 5]
```diff
@@ -55,31 +55,78 @@ extern "C" fn wraper_fn(
...
     pub(crate) ctx: *mut wasmedge::WasmEdge_FunctionInstanceContext,
     pub(crate) registered: bool,
     pub(crate) ty: Option<FuncType>,
 }
-
 impl Function {
     /// wasmedge::WasmEdge_FunctionInstanceCreate take C functions
     /// This may not be implement, base on the limiation of passing C functions in Rust lang.
-    /// Please refer [`create_bindings`] for building hostfunctions.
+    /// Please refer `create_bindings` for building host functions.
```
Comment (hydai) [inline bindings/rust/wasmedge-sys/src/instance/function.rs:74]: `refer to`

**Review by hydai — CHANGES_REQUESTED** [review]

File: bindings/rust/wasmedge-sys/src/instance/function.rs (hunk 5 above)
Comment (apepkuss) [reply bindings/rust/wasmedge-sys/src/instance/function.rs:74]: Accepted. I'll fix the issue. Thanks!

**Comment by apepkuss** [comment]
@hydai, the issue has been fixed, please check again. Thanks!

**Review by hydai — APPROVED** [review]

**Pull request merged**